notify.go

```diff
@@ -801,28 +797,22 @@ func parseNewTicketsNtfnParams(params []json.RawMessage) (*chainhash.Hash,
 		return nil, 0, 0, nil, err
 	}
 
-	// Unmarshal fourth parameter as a map[hash]hash.
-	tickets := make(map[string]string)
+	// Unmarshal fourth parameter as a slice.
+	tickets := make([]string, 0)
 	err = json.Unmarshal(params[3], &tickets)
 	if err != nil {
 		return nil, 0, 0, nil, err
 	}
-	t := make(map[chainhash.Hash]chainhash.Hash)
-
-	for hashStr, ticketNH := range tickets {
-		// Create and cache ShaHash from tx hash.
-		ticketSha, err := chainhash.NewHashFromStr(hashStr)
-		if err != nil {
-			return nil, 0, 0, nil, err
-		}
+	t := make([]*chainhash.Hash, len(tickets))
 
-		// Convert the numberData to a big int.
-		numberHash, err := chainhash.NewHashFromStr(ticketNH)
+	for _, ticketHashStr := range tickets {
+		// Create and cache Hash from tx hash.
+		ticketHash, err := chainhash.NewHashFromStr(ticketHashStr)
 		if err != nil {
 			return nil, 0, 0, nil, err
 		}
 
-		t[*ticketSha] = *numberHash
+		t = append(t, ticketHash)
```

Comment:
This will start appending new hashes to a slice of a bunch of nil pointers.  Either need to assign t[i] or make the slice with 0 len.